eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -1891,4 +1893,151 @@ public function copyRelations($originalContentId, $copiedContentId, $versionNo =
+        $this->connection->beginTransaction();
+        try {
+            $this->deleteTranslationFromContentVersions($contentId, $language->id);
+            $this->deleteTranslationFromContentObject($contentId, $language->id);
+
+            $this->deleteTranslationFromContentAttributes($contentId, $languageCode);
+            $this->deleteTranslationFromContentNames($contentId, $languageCode);
+
+            $this->connection->commit();
+        } catch (DBALException $e) {
+            $this->connection->rollBack();
+            throw $e;
+        }
+    }
+
+    /**
+     * Delete translation from the ezcontentobject_attribute table.
+     *
+     * @param int $contentId
+     * @param string $languageCode
+     */
+    private function deleteTranslationFromContentAttributes($contentId, $languageCode)
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->delete('ezcontentobject_attribute')
+            ->where('contentobject_id = :contentId')
+            ->andWhere('language_code = :languageCode')
+            ->setParameters(
+                [
+                    ':contentId' => $contentId,
+                    ':languageCode' => $languageCode,
+                ]
+            )
+        ;
+
+        $query->execute();
+    }
+
+    /**
+     * Delete translation from the ezcontentobject_name table.
+     *
+     * @param $contentId
+     * @param $languageCode
+     */
+    private function deleteTranslationFromContentNames($contentId, $languageCode)
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->delete('ezcontentobject_name')
+            ->where('contentobject_id=:contentId')
+            ->andWhere('real_translation=:languageCode')
+            ->setParameters(
+                [
+                    ':languageCode' => $languageCode,
+                    ':contentId' => $contentId,
+                ]
+            )
+        ;
+
+        $query->execute();
+    }
+
+    /**
+     * Remove language from language_mask of ezcontentobject.
+     *
+     * @param int $contentId
+     * @param int $languageId
+     * @throws \eZ\Publish\Core\Base\Exceptions\BadStateException
+     */
+    private function deleteTranslationFromContentObject($contentId, $languageId)
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query->update('ezcontentobject')
+            // parameter for bitwise operation has to be placed verbatim (w/o binding) for this to work cross-DBMS
+            ->set('language_mask', 'language_mask & ~ ' . $languageId)
+            ->set('modified', ':now')
+            ->where('id = :contentId')
+            ->andWhere(
+            // make sure removed translation is not the last one (incl. alwaysAvailable)
+                $query->expr()->orX(
+                    'language_mask & ~ ' . $languageId . ' <> 0',
+                    'language_mask & ~ ' . $languageId . ' <> 1'
+                )
```

Comment:
Note: More bitwise voodoo. If the above operation gives us either 0 or 1 this means the language was the last one (1 - alwaysAvailable). There's a small risk of race between validation in Core `ContentService::removeTranslation` and DBMS transaction, so I placed this condition just to be safe.